packages/babel-helpers/src/helpers.js

```diff
@@ -470,6 +470,16 @@ helpers.wrapNativeSuper = defineHelper(`
   }
 `);
 
+helpers.constructSuperInstance = defineHelper(`
+  export default function _constructSuperInstance(Derived, args, existingThis) {
+    const Super = Derived.__proto__ || Object.getPrototypeOf(Derived);
+    return typeof Reflect !== "undefined" && typeof Reflect.construct === "function" 
```

Comment:
While checking if my assumptions were really correct, I found that, while Node 4 has an (incomplete and almost unusable) implementation of classes, it also does not have `Reflect`, at all.

However, all that changes with this helper is that nothing changes on Node 4, you still can't extend native classes 😄

I also checked whether spreading `arguments` was safe; while Node 4 has no support for spread at all, `arguments` _does_ have a working `Symbol.iterator`, so we should be safe on that side as well if the user uses `transform-spread`.